src/backend/tests/unit/services/tracing/test_tracing_service.py

```diff
@@ -0,0 +1,482 @@
+    settings = Settings()
+    settings.deactivate_tracing = False
+    return SettingsService(settings, MagicMock())
+
+
+@pytest.fixture
+def tracing_service(mock_settings_service):
+    return TracingService(mock_settings_service)
+
+
+@pytest.fixture
+def mock_component():
+    component = MagicMock()
+    component._vertex = MagicMock()
+    component._vertex.id = "test_vertex_id"
+    component.trace_type = "test_trace_type"
+    return component
+
+
+@pytest.mark.asyncio
+async def test_start_end_tracers(tracing_service):
+    """Test starting and ending tracers."""
+    run_id = uuid.uuid4()
+    run_name = "test_run"
+    user_id = "test_user"
+    session_id = "test_session"
+    project_name = "test_project"
+    outputs = {"output_key": "output_value"}
+
+    with patch("langflow.services.tracing.service._get_langsmith_tracer", return_value=MockTracer), \
+         patch("langflow.services.tracing.service._get_langwatch_tracer", return_value=MockTracer), \
+         patch("langflow.services.tracing.service._get_langfuse_tracer", return_value=MockTracer), \
+         patch("langflow.services.tracing.service._get_arize_phoenix_tracer", return_value=MockTracer):
+
+        await tracing_service.start_tracers(run_id, run_name, user_id, session_id, project_name)
+        # Verify trace_context is set correctly
+        trace_context = trace_context_var.get()
+        assert trace_context is not None
+        assert trace_context.run_id == run_id
+        assert trace_context.run_name == run_name
+        assert trace_context.project_name == project_name
+        assert trace_context.user_id == user_id
+        assert trace_context.session_id == session_id
+
+        # Verify tracers are initialized
+        assert "langsmith" in trace_context.tracers
+        assert "langwatch" in trace_context.tracers
+        assert "langfuse" in trace_context.tracers
+        assert "arize_phoenix" in trace_context.tracers
+
+        await tracing_service.end_tracers(outputs)
+
+        # Verify end method was called for all tracers
+        trace_context = trace_context_var.get()
+        for tracer in trace_context.tracers.values():
+            assert tracer.end_called
+            assert tracer.metadata_param == outputs
+            assert tracer.outputs_param == trace_context.all_outputs
+
+        # Verify worker_task is cancelled
+        assert trace_context.worker_task is None
+        assert not trace_context.running
+
+
+@pytest.mark.asyncio
+async def test_trace_component(tracing_service, mock_component):
+    """Test component tracing context manager."""
+    run_id = uuid.uuid4()
+    run_name = "test_run"
+    user_id = "test_user"
+    session_id = "test_session"
+    project_name = "test_project"
+
+    trace_name = "test_component_trace"
+    inputs = {"input_key": "input_value"}
+    metadata = {"metadata_key": "metadata_value"}
+
+    with patch("langflow.services.tracing.service._get_langsmith_tracer", return_value=MockTracer), \
```

Comment:
I'm unable to mark the other lines through GitHub's interface.
But, this entire block needs to be indented as follows:
```suggestion
    with (
        patch("langflow.services.tracing.service._get_langsmith_tracer", return_value=MockTracer),
        patch("langflow.services.tracing.service._get_langwatch_tracer", return_value=MockTracer),
        patch("langflow.services.tracing.service._get_langfuse_tracer", return_value=MockTracer),
        patch("langflow.services.tracing.service._get_arize_phoenix_tracer", return_value=MockTracer),
    ):
```